server.js

```diff
@@ -6030,6 +6030,133 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Uptime Robot status integration.
+// API documentation : https://uptimerobot.com/api
+camp.route(/^\/uptimerobot\/status\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var monitorApiKey = match[1];  // eg, u956-afus321g565fghr519
+  var format = match[2];
+  var badgeData = getBadgeData('status', data);
+  var options = {
+    method: 'POST',
+    json: true,
+    body: {
+      "api_key": monitorApiKey,
+      "format": "json"
+    },
+    uri: 'https://api.uptimerobot.com/v2/getMonitors'
+  };
+  // A monitor API key must start with "m"
+  if (monitorApiKey.substring(0, "m".length) !== "m") {
+    badgeData.text[1] = 'must use a monitor key';
+    sendBadge(format, badgeData);
+    return;
+  }
+  request(options, function(err, res, json) {
+    if (err !== null || res.statusCode >= 500 || typeof json !== 'object') {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      if (json.stat === 'fail') {
+        badgeData.text[1] = 'vendor error';
+        if (json.error && typeof json.error.message === 'string') {
+          badgeData.text[1] = json.error.message;
+        }
+        badgeData.colorscheme = 'lightgrey';
+        sendBadge(format, badgeData);
+        return;
+      }
+      var status = json.monitors[0].status;
+      if (status === 0) {
+        badgeData.text[1] = 'paused';
+        badgeData.colorscheme = 'yellow';
+      } else if (status === 1) {
+        badgeData.text[1] = 'not checked yet';
+        badgeData.colorscheme = 'yellowgreen';
+      } else if (status === 2) {
+        badgeData.text[1] = 'up';
+        badgeData.colorscheme = 'brightgreen';
+      } else if (status === 8) {
+        badgeData.text[1] = 'seems down';
+        badgeData.colorscheme = 'orange';
+      } else if (status === 9) {
+        badgeData.text[1] = 'down';
+        badgeData.colorscheme = 'red';
+      } else {
+        badgeData.text[1] = 'invalid';
+        badgeData.colorscheme = 'lightgrey';
+      }
+      sendBadge(format, badgeData);
+    } catch(e) {
+      badgeData.text[1] = 'invalid';
+      sendBadge(format, badgeData);
+    }
+  });
+}));
+
+// Uptime Robot ratio integration.
+// API documentation : https://uptimerobot.com/api
+camp.route(/^\/uptimerobot\/ratio\/(.*)\/(.*)\.(svg|png|gif|jpg|json)$/,
```

Comment:
In testing this out, I found it was easy to miss the `numberOfDays` parameter. What do you think about moving it before the key, so it's less easily missed, and making it optional?

```
http://img.shields.io/uptimerobot/ratio/m778918918-3e92c097147760ee39d02d36.svg
http://img.shields.io/uptimerobot/ratio/30/m778918918-3e92c097147760ee39d02d36.svg
```